### DISCUSSION TITLE
Using Datafeedr to update stock levels and affiliate ID of existing products

### DISCUSSION THREAD

**Discussion by buffdownunder** [datafeedr/wordpress-plugins#32] [Help] [answered]
Hi,

I just came across DatafeedR and love it.
I'm definitely going to use it for a website I have in mind.

Yet I also want to use it for an existing website I have.

That website has been selling one brand's products for over 18 years.  I now want to transition it from "I have products in stock and ship" to "I list everything even if I don't have it in stock."

The issue with this brand on the web is that the listing of their products/designs is a total mess. Some designs are listed as single products, and others as variation products. So importing products will result in messy listings and category archive pages.

My value proposition would be to organise the mess by providing the entire product catalogue available worldwide (there are differences between countries, like the national flag designs).
I would also provide guides and form to help select the correct products and other value-added information that meets Google's E-E-A-T principle.

But this all means that I wouldn't import products. I would create them myself.

I would require DatafeedR to create Comparison sets for a product based on an identifier.
Could DatafeedR do this?

Regards,
Edward

**Comment by EricBusch** [comment, marked as answer]
Hi Edward

Thanks for reaching out!

Yes, Datafeedr can create Comparison Sets based on identifiers such as UPC or EAN.

We've outlined how to do that here: https://datafeedrapi.helpscoutdocs.com/article/297-datafeedr-price-comparison-shortcodes

Hope this helps!

If you have any other questions, please don't hesitate to ask.

Thanks,
Eric

**Comment by buffdownunder** [comment]
Is there a way to test the search results? My current issue is that there are many retailers and affiliate programs.
I need to get a feeling of how many programs I have to join to display the entire brand catalogue. What is the best way to start in your opinion, Eric?

**Comment by EricBusch** [comment]
Hi

I think the best way to start would be to search for the products you are interested in promoting in our Scratchpad: https://datafeedr.me/scratchpad

Then you'll be able to identify which networks and merchants sell those products.

Here's more info: https://datafeedrapi.helpscoutdocs.com/article/302-how-to-use-the-scratchpad

Hope this helps!

Eric

**Reply by buffdownunder** [reply]
It does, Eric. The Scratchpad is incredible.

Yet it throws up some questions that puzzle me in my initial search:

A: I understand that you can't import Amazon Products using DatafeedR. That you have to use another plugin.
Yet I see Amazon used in Screenshots in comparison sets. I assume that DatafeedR can still get prices and availability even though Amazon is not on the list of merchants in the search results.
Is this correct?
I also don't see eBay. Amazon and eBay have affiliate programs and are very popular in Australia.

B: A product only seems on special if it is marked "on special" on the merchant's site. I see that one can search for prices under a manually set price.
But can DatafeedR display the price of the Woocommerce product as on special if the price is under the product's list price? I assume not, as this would require a check of a list price before deciding where to write the merchants' price.
Please have a look at the screenshot to see what I mean. The products shown are the same, but with different design and size. One is marked as "on special" in the JSON data, and the other is not. The list price is the same for these two products—just one example of how inadequate and inconsistent the listings of this brand's products are. The example is the same merchant. Go figure. It blows my mind how this can happen.

C: Now I understand that you can't build a plugin that caters to all the bad product listings and inconsistencies.
Affiliate marketing isn't also focusing on cleaning up the mess.
That's just my value proposition. The scratchpad results are nothing short of extraordinary.

What I could see as my solution is to create a non-indexed woocommerce installation that imports all products as shown on the scratchpad and then updates the live site based on an analysis of all the products.
It wouldn't be rocket science for me to design SQL queries to do that.
I already do quite a lot outside the woocommerce dB because it's easier than building a plugin and doesn't add to the server load.
Would your terms cover this?
Or do you have a developers' area to create custom outputs? Taking the specials topic as an example to set the scratchpad's search results on special if the price is under a manually set price for example.

Kind regards,

Edward
![Screen Shot 2023-02-13 at 9 04 28 pm](https://user-images.githubusercontent.com/42963473/218563413-b449c8b7-b0fa-4647-b699-bcb0fed99f28.jpg)

**Reply by buffdownunder** [reply]
Here's the scratchpad's url I used. Have a look and you will see just by scanning the titles how inconsistent the product listings are.
https://scratchpad.datafeedr.com?expires=1678910160&query=YTozOntzOjU6ImJyYW5kIjtzOjQ6ImJ1ZmYiO3M6ODoiY3VycmVuY3kiO3M6MzoiVVNEIjtzOjc6Imluc3RvY2siO3M6MzoieWVzIjt9&signature=f9b14aa180c6650fd117180702510b4a58fbc267c09ecd253c8e5fea8905b7f2
I'm tempted now to look at using your Scratchpads results and GPT3 to create standardised product descriptions based on the product descriptions and images of all merchants :-)

**Comment by EricBusch** [comment]
Hi

A. Yes, our Comparison Sets plugin is integrated with Amazon so it can display products from Amazon.

B. When you say "on special" do you mean "on sale"? Our plugin will handle the importing of regularly priced products as well as handle importing the sale price of a product when importing a product.

If there is another "special" status for a product we are not aware of that and do not support it.

C. We don't disallow you to import products into your WooCommerce store and then run further queries against that data. However, if you are manipulating the product data and overriding the data provided by the merchant, you would have to check with the terms of agreement from the merchants whose data you are modifying to see if this is allowed. My guess is that this is not allowed. But you should seek clarification from the affiliate networks and the merchants.

Hope this helps!
Eric

**Reply by buffdownunder** [reply]
Yes, it does, Eric. Thanks.
Now last but not least, something I couldn't figure out.
How do I estimate API Calls?
If I take the scratchpad URL from my other reply and say, I want to display all these products. How many API calls would this be?
Is there something like an API call calculator?

Regards,
Edward

**Comment by EricBusch** [comment]
Hi

If you are making calls directly to our API programmatically, each request can return up to 100 products before you have to "paginate" to the next 100 products.

So, if your above Scratchpad search returned 850 results, it will require 9 API requests to access all 850 records:

- Request 1 - 1-100 records
- Request 2 - 101-200 records
- Request 3 - 201-300 records
- etc...
- Request 8 - 701-800 records
- Request 9 - 801-850 records

Hope this helps!

Eric
